### PR TITLE
test: mock popular searches adapter in e2e tests

### PR DESCRIPTION
**Pull request by mavmaf**
EX-4466

## Motivation and Context ##

We needed to have a reliable method of doing e2e tests without relying on APIs outside of our control. This PR implements a version of that solution for the Popular Searches.

- mocked-adapter changed so the getPopularSearches can be intercepted by Cypress
- e2e test for the Popular Searches changed to make use of Cypress' intercept
- Scenario 2 of Popular Searches tests was simplified, together with its .vue, in order to skip checking other components behavior (NQs, RTs, QSs and Results) since it's already being tested in the Search tests (although its refactoring to avoid using the real API is still pending).

## What is the destination branch of this PR? ##

*EX-4466*

## How Has This Been Tested? ##

Run the tests in Cypress to see the intercept working.
In `cypress.json` you can change the testFiles property's first element to `"cucumber/**/*.features"` to be able to select the test to run.

### CODE REVIEW

File: packages/x-components/src/__stubs__/suggestions-stubs.factory.ts [hunk 1]
```diff
@@ -1,43 +1,37 @@
...
  *
- * @param modelName - Model name for each suggestion.
- * @returns Array of suggestion stub.
+ * @param amount - Number of stubbed suggestions to create.
+ *
+ * @returns Array of suggestions stub.
+ *
+ * @internal
+ */
+export function getSuggestionStub(amount = 3): Suggestion[] {
```
Comment (pmareke) [inline packages/x-components/src/__stubs__/suggestions-stubs.factory.ts:12]: This change from `getSuggestionsStub` to `getSuggestionStub` breaks some tests, please take a look.

File: packages/x-components/tests/e2e/cucumber/popular-searches/popular-searches.spec.ts [hunk 2]
```diff
@@ -46,7 +59,7 @@ When('popular search number {int} is clicked', (popularSearchItem: number) => {
 });
 
 And(
-  'the clicked popular search is removed from Popular Searches if {boolean} is true',
+  'the clicked popular search is removed from Popular Searches if hideSessionQueries = {boolean} is true',
```
Comment (pmareke) [inline packages/x-components/tests/e2e/cucumber/popular-searches/popular-searches.spec.ts:62]: Please fix this error

image

**Review by pmareke — CHANGES_REQUESTED** [review]
Fix the build and all the errors please.

**Review by davidmfempathy — APPROVED** [review]

File: packages/x-components/src/__stubs__/suggestions-stubs.factory.ts [hunk 3]
```diff
@@ -41,3 +41,24 @@ export function getSuggestionsStub(modelName: ModelNameType): Suggestion[] {
...
+ * @returns A suggestion.
+ */
+export function createSuggestionStub(query: string, suggestion?: Partial<Suggestion>): Suggestion {
+  return {
+    facets: [],
+    key: 'key',
+    query,
+    totalResults: 10,
+    results: [],
+    modelName: 'PopularSearch',
```
Comment (pmareke) [inline packages/x-components/src/__stubs__/suggestions-stubs.factory.ts:61]: If we're creating suggestions why are we using always the `PopularSearch` modelName?

**Review by pmareke — CHANGES_REQUESTED** [review]

**Review by pmareke — APPROVED** [review]

File: packages/x-components/tests/e2e/cucumber/empathize.feature [hunk 4]
```diff
@@ -2,6 +2,7 @@ Feature: Full Empathize
 
   Background:
     Given a next queries API
+    Given a suggestions API
```
Comment (pmareke) [inline packages/x-components/tests/e2e/cucumber/empathize.feature:5]: I'm not sure why are we adding this `Given`

File: packages/x-components/tests/e2e/cucumber/history-queries.feature [hunk 5]
```diff
@@ -2,6 +2,7 @@ Feature: History queries component
 
   Background:
     Given a next queries API
+    Given a suggestions API
```
Comment (pmareke) [inline packages/x-components/tests/e2e/cucumber/history-queries.feature:5]: I'm not sure why are we adding this Given

File: packages/x-components/tests/e2e/cucumber/search-box.feature [hunk 6]
```diff
@@ -2,6 +2,7 @@ Feature: Search-box component
 
   Background:
     Given a next queries API
+    Given a suggestions API
```
Comment (pmareke) [inline packages/x-components/tests/e2e/cucumber/search-box.feature:5]: I'm not sure why are we adding this Given

**Review by pmareke — CHANGES_REQUESTED** [review]

**Review by pmareke — APPROVED** [review]

**Review by davidmfempathy — APPROVED** [review]

**Pull request merged**